rpmio/rpmlua.cc

```diff
@@ -1365,19 +1365,37 @@ static int luaopen_rpm(lua_State *L)
     return 1;
 }
 
-void check_deprecated(lua_State *L, const char *func, const char *deprecated_in)
+void check_deprecated(lua_State *L, const char *func,
+		      const char *deprecated_in, const char *removed_in)
 {
-    int warn = 1;
+    int err = 1;
+    const char *rpmversion = NULL;
     lua_getfield(L, LUA_REGISTRYINDEX, "RPM_PACKAGE_RPMVERSION");
-    if (lua_isstring(L, -1)) {
-	rpmver v1 = rpmverParse(lua_tostring(L, -1));
+    if (lua_isstring(L, -1))
+	rpmversion = lua_tostring(L, -1);
+    if (rpmversion) {
+	rpmver v1 = rpmverParse(rpmversion);
+	rpmver v2 = rpmverParse(removed_in);
+	if (v2 && rpmverCmp(v1, v2) < 0)
+	    err = 0;
+	rpmverFree(v2);
+	rpmverFree(v1);
+    }
+    lua_pop(L, 1);
+
+    if (err) {
+	luaL_error(L, "%s() is no longer available, use rpm.spawn() or rpm.execute() instead", func);
+    }
+
+    int warn = 1;
+    if (rpmversion) {
+	rpmver v1 = rpmverParse(rpmversion);
```

Comment:
This repeats the parsing from line 1377.

Reply:
Yup. The reason it's written this way is the luaL_error() return from the middle - we need allocations freed up before luaL_error() might get called.